docs/decisions/managing-service-dependencies.md

```diff
@@ -0,0 +1,86 @@
+# Managing Service Dependencies
+
+## Context and Problem Statement
+
+As we split OpenFoodFacts up into smaller, more manageable, services, it becomes increasingly difficult for developers to know what additional services need to be deployed in order for the service they are working on to function correctly. This becomes increasingly difficult if the referenced service itself has other dependencies.
+
+Note that for development and testing purposes it is recommended that any dependencies are mocked or managed using tools like testcontainers, rather than relying on a separate service being up. This document concerns itself more with creating a workable end-to-end deployment that can be used for ad-hoc, interactive testing and discovery.
+
+This document is specifically concerned with "internal" dependencies where shared state is needed, e.g. each service can have its own independent database rather than relying on a shared database service, but the messaging system (Redis) needs to be shared in order for messages to pass between services. Dependencies on services outside the scope of OpenFoodFacts are referred to as "external" dependencies.
+
+## Decision Drivers
+
+* the definition of dependencies should be as close as possible to the particular service's source code
+* existing tooling should be used wherever possible
+* the mechanism to pull in any dependencies should be as simple and as fast as possible
+* any mechanism needs to be compatible with all supported development environments, e.g. local code, dev containers, GitPod
+* it should be relatively straightforward for a developer to switch from a simple "latest" deployment of a referenced service to a version that is also undergoing local development
+
+## Considered Options
+
+* Use the docker compose include command to reference dependencies in sibling folders
+* Develop a custom tool that works from a list of dependencies defined in a file
+
+## Decision Outcome
+
+Custom tooling will be used so that we can cope with circular dependencies and to allow loading of dependencies to be performed using a single command.
+
+Each project must provide the following:
+
+* A file with the list of other services it depends on
+* a docker compose file which loads the service and any of its external dependencies from the latest available images
```

Comment:
I think this image version should be configurable (in case your devs impact two repositories) but should defaults to latest stable version. 

It can be done through environment, of course.

Reply:
```suggestion
* a docker compose file which loads the service and any of its external dependencies (from the latest stable images by default, but configurable through environment)
```